node_modules/oae-principals/lib/internal/dao.js

```diff
@@ -961,3 +1034,130 @@ var getAllUsersForTenant = module.exports.getAllUsersForTenant = function(tenant
+        if (err) {
+            return callback(err);
+        } else if (_.isEmpty(rows)) {
+            return callback(null, []);
+        }
+
+        var userArchive = _.chain(rows).map(_rowToArchive);
+        return callback(null, userArchive._wrapped[0]);
+    });
+};
+
+/**
+ * Add datas in user archive
+ *
+ * @param  {String}         archiveId           The archive id ot the principal tenant 
+ * @param  {String}         principalId         The principal id of the removed user
+ * @param  {String}         resourceId         The resource id of the removed user
+ * @param  {Function}       callback            Standard callback function
+ * @param  {Object}         callback.err        An error that occurred, if any
+ * @param  [User]           callback.users      The list of users for the given tenancy
+ */
+var addDataToArchive = module.exports.addDataToArchive = function(archiveId, principalId, resourceId, callback) {
+
+    var queries = [];
+    var stringResource = '';
+    stringResource = resourceId.join();
+
+    // Prepare query
+    var query = 'INSERT INTO "DataArchive" ("archiveId", "principalId", "resourceId") VALUES (?, ?, ?)';
+    var parameters = [archiveId, principalId, stringResource];
+
+    // Run query
+    Cassandra.runQuery(query, parameters, function (err) {
+        if (err) {
+            return callback(err);
+        }
+        return callback(null);
+    });
+};
+
+/**
+ * Remove datas from user archive
+ *
+ * @param  {String}         archiveId           The archive id ot the principal tenant 
+ * @param  {String}         principalId         The principal id of the removed user
+ * @param  {Function}       callback            Standard callback function
+ * @param  {Object}         callback.err        An error that occurred, if any
+ * @param  [User]           callback.users      The list of users for the given tenancy
+ */
+var removeDataInArchive = module.exports.removeDataInArchive = function(archiveId, principalId, callback) {
+
+    Cassandra.runQuery('DELETE FROM "DataArchive" WHERE "archiveId" = ? AND "principalId" = ?', [archiveId, principalId], function(err, rows) {
+        if (err) {
+            return callback(err);
+        }
+        return callback(null);
+    });
+
+};
+
+/**
+ * Get a user archive from a tenant 
+ *
+ * @param  {String}         alias                   The tenant alias
+ * @param  {Function}       callback                Standard callback function
+ * @param  {Object}         callback.err            An error that occurred, if any
+ * @param  [UserArchive]    callback.userArchive    The archive id ot the tenant
+ */
+var getArchivedUser = module.exports.getArchivedUser = function(alias, callback) {
+
+    Cassandra.runQuery('SELECT * FROM "ArchiveByTenant" WHERE "tenantAlias" = ?', [alias], function(err, rows) {
+        if (err) {
+            return callback(err);
+        } 
+        var userArchive = _.chain(rows).map(_rowToArchive);
```

Comment:
Why are we using _.chain here? Also, usually when trying to get the wrapped value with underscore, one can use `_.value` - in this case it'd be `_.chain(rows).map(_rowToArchive).value();` I think (although I'm pretty sure you could just go with `_.map(rows, _rowToArchive);`).